doc: update how-to, reference landing pages (#1348)
canonical/microcloud: doc/explanation/index.md

---
myst:
  html_meta:

>     description: An index of deep-dive explanations that introduce you to concepts in MicroCloud, including its relationship to LXD, Ceph, and OVN and its approach to networking, storage, and security.

Reviewer: ```suggestion
    description: An index of explanations about MicroCloud concepts, such as its relationship to LXD, Ceph, and OVN and its approach to networking, storage, and security.
```

I should have mentioned — we want to keep important bits within the first 150-160 chars to prevent being truncated by search engines in the search results snippet.

Author: Updated the three pages you flagged
